data-plane/dispatcher-loom/src/main/java/dev/knative/eventing/kafka/broker/dispatcherloom/LoomKafkaConsumer.java

```diff
@@ -76,43 +79,57 @@ public Future<Map<TopicPartition, OffsetAndMetadata>> commit(Map<TopicPartition,
         final Promise<Map<TopicPartition, OffsetAndMetadata>> promise = Promise.promise();
         addTask(
                 () -> {
-                    consumer.commitAsync(offset, (offsetMap, exception) -> {
-                        if (exception != null) {
-                            promise.fail(exception);
-                        } else {
-                            promise.complete(offsetMap);
-                        }
-                    });
+                    try {
+                        consumer.commitSync(offset);
```

Comment:
the switch to sync is needed because?

Reply:
In the docs, the ordering of the async calls is not clear for everything that is not `commit*` meaning that it's not clear (from the docs) what happens when calling:
```
commitAsync()
close()
```

will close wait for commitAsync to return ?

So, better be safe here for now